kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/fragment/FeedbackInputFragment.java

```diff
@@ -0,0 +1,215 @@
+                getFragmentManager().popBackStack();
+            }
+        });
+
+        buttonCloseFragment.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                getFragmentManager().popBackStack();
+            }
+        });
+
+        //set the un-selected size of the buttons to 80% of size
+        buttonsRating[0].setScaleX(0.8f);
+        buttonsRating[0].setScaleY(0.8f);
+        buttonsRating[1].setScaleX(0.8f);
+        buttonsRating[1].setScaleY(0.8f);
+        buttonsRating[2].setScaleX(0.8f);
+        buttonsRating[2].setScaleY(0.8f);
+
+        buttonsRating[0].setOnClickListener(this);
+        buttonsRating[1].setOnClickListener(this);
+        buttonsRating[2].setOnClickListener(this);
+
+        return view;
+    }
+
+    /**
+     * toggle button as selected or deselected
+     * @param select true or false
+     * @param b the button to toggle
+     */
+    void toggleRatingButtonSelected(boolean select, Button b) {
+        //if show the feedback comment input edit text, if not already visible
+        if(editTextFeedbackComment.getVisibility()==View.GONE) {
+            editTextFeedbackComment.setVisibility(View.VISIBLE);
+        }
+
+        //default initialization
+        TextView t = textViewsRatingText[1];
+
+        int buttonId = b.getId();
+
+        //set drawable
+        if (buttonId == R.id.idButtonPoor) {
+            t = textViewsRatingText[0];
+            if(select)
+                b.setBackground(getResources().getDrawable(R.drawable.ic_sad_1));
+            else
+                b.setBackground(getResources().getDrawable(R.drawable.ic_sad_1_grey));
+        }
+        else if (buttonId == R.id.idButtonAverage) {
+            t = textViewsRatingText[1];
+            if(select)
+                b.setBackground(getResources().getDrawable(R.drawable.ic_confused));
+            else
+                b.setBackground(getResources().getDrawable(R.drawable.ic_confused_grey));
+        }
+        else if (buttonId == R.id.idButtonGood) {
+            t = textViewsRatingText[2];
+            if(select)
+                b.setBackground(getResources().getDrawable(R.drawable.ic_happy));
+            else
+                b.setBackground(getResources().getDrawable(R.drawable.ic_happy_grey));
+        }
+
+        //set visibility and size
+        if(select) {
+            b.setScaleX(1f);
+            b.setScaleY(1f);
+            t.setVisibility(View.VISIBLE);
+        } else {
+            b.setScaleX(0.8f);
+            b.setScaleY(0.8f);
+            t.setVisibility(View.GONE);
+        }
+    }
+
+    //when one of the three rating buttons are clicked
+    @Override
+    public void onClick(View view) {
+        int buttonId = view.getId();
+
+        //counter variable for setting rating value
+        int i = 0;
+        for(Button button : buttonsRating) {
+            if(button.getId()==buttonId) {
```

Comment:
@shubhamtewari does android provide a way to set some attribute on element and fetch the value directly from it, instead of looping and comparing the button id?

Reply:
yes, implemented it